src/convolution_api.cpp

```diff
@@ -113,12 +114,14 @@ static inline auto MakeWrWCtxAndProblem(miopenHandle_t handle,
     return std::make_tuple(std::move(ctx), std::move(problem));
 }
 
+MIOPEN_EXPORT
```

Comment:
Why do the `.cpp` files need `MIOPEN_EXPORT` attribute? Doesn't it suffice to just put it on the headers (internal or external, i.e., stuff under `src/include`).

Reply:
There is no header file providing the prototype for that function.

Reply:
Doesn't that mean that we don't need to export these symbols? Because the only way to use such a symbol would be through a public or internal header.

Reply:
If you see `MIOPEN_EXPORT` or `MIOPEN_INTERNALS_EXPORT` in the header or source file, that means we need to export those symbols because something (e.g., MIOpenDriver, examples, tests, etc.) is using them. All of the exports you see are relevant. Any of them missing will cause the Windows build to fail to link. There will be more in the future.

Reply:
To be more precise, you don't need a header file declaring exported functions to use them in your code. You may put a function declaration in your code, and the linker will pick it up automatically from the list of exported functions of a certain library that has that function exported.